wakuv2/waku.go

```diff
@@ -1038,43 +933,41 @@ func (w *Waku) Subscribe(f *common.Filter) (string, error) {
 		f.PubsubTopic = relay.DefaultWakuTopic
 	}
 
-	s, err := w.filters.Install(f)
+	id, err := w.filters.Install(f)
 	if err != nil {
-		return s, err
+		return id, err
 	}
 
 	if w.settings.LightClient {
-		go func() {
-			ticker := time.NewTicker(1 * time.Second)
-			for range ticker.C {
-				err := w.subscribeToFilter(f)
-				if err == nil {
-					break
-				}
-			}
-		}()
+		w.filterManager.eventChan <- FilterEvent{eventType: FilterEventAdded, filterID: id}
 	}
 
-	return s, nil
+	return id, nil
 }
 
 // Unsubscribe removes an installed message handler.
 func (w *Waku) Unsubscribe(ctx context.Context, id string) error {
-	f := w.filters.Get(id)
-	if f != nil && w.settings.LightClient {
-		contentFilter := w.buildContentFilter(f.PubsubTopic, f.Topics)
-		if _, err := w.node.FilterLightnode().Unsubscribe(ctx, contentFilter); err != nil {
-			return fmt.Errorf("failed to unsubscribe: %w", err)
-		}
-	}
-
 	ok := w.filters.Uninstall(id)
 	if !ok {
 		return fmt.Errorf("failed to unsubscribe: invalid ID '%s'", id)
 	}
+
+	if w.settings.LightClient {
+		w.filterManager.eventChan <- FilterEvent{eventType: FilterEventRemoved, filterID: id}
```

Comment:
This looks suspicious, as it might end up in a panic if the channel is closed, I'll look at the code to make sure, just a note for now

Reply:
Channel is not closed anywhere.

Reply:
it's fine not to close a channel if you think it will be recycled by the garbage collector, we probably want to figure out the lifecycle, just to make sure, otherwise we'd have a memory leak

Reply:
Added a buffer to `eventChan` to address an issue of goroutines hanging around.